clients/lsp-rust.el

```diff
@@ -665,9 +696,13 @@ https://rust-analyzer.github.io/manual.html#auto-import.
             :unsetTest ,lsp-rust-analyzer-cargo-unset-test)
     :rustfmt (:extraArgs ,lsp-rust-analyzer-rustfmt-extra-args
               :overrideCommand ,lsp-rust-analyzer-rustfmt-override-command)
-    :inlayHints (:typeHints ,(lsp-json-bool lsp-rust-analyzer-server-display-inlay-hints)
+    :inlayHints (:renderColons ,(lsp-json-bool lsp-rust-analyzer-server-format-inlay-hints)
+                 :typeHints ,(lsp-json-bool lsp-rust-analyzer-server-display-inlay-hints)
                  :chainingHints ,(lsp-json-bool lsp-rust-analyzer-display-chaining-hints)
+                 :closureReturnTypeHints ,(lsp-json-bool lsp-rust-analyzer-display-closure-return-type-hints)
+                 :lifetimeElisionHints ,lsp-rust-analyzer-display-lifetime-elision-hints
```

Comment:
I think the structure of this setting might change again soon (right, @Veykril ?), so maybe it's best to just leave it out

Reply:
I believe you're right, that's rust-analyzer/rust-analyzer#11778.  Hasn't actually happened yet, but I'm aware it almost certainly will.

Reply:
I went ahead and changed one of the configs here https://github.com/rust-analyzer/rust-analyzer/pull/11789/files

Note that r-a might have to change a bunch of configs in the future for consistency ( we dug ourselves a bad hole here https://github.com/rust-analyzer/rust-analyzer/issues/11790 )